heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -97,60 +101,152 @@ const JEST_CONFIG_PACKAGE_FOLDER: string = path.dirname(require.resolve('jest-co
+        passWithNoTests: true,
+        silent: silent.value || pluginOptions?.silent,
+        testNamePattern: testNamePattern.value || pluginOptions?.testNamePattern,
+        testPathPattern: testPathPattern.value || pluginOptions?.testPathPattern,
+        testTimeout: testTimeout.value ? parseInt(testTimeout.value, 10) : pluginOptions?.testTimeout,
+        updateSnapshots: updateSnapshots.value || pluginOptions?.updateSnapshots
+      };
+      await this._runJestAsync(taskSession, heftConfiguration, combinedOptions);
+    });
+  }
+
+  /**
+   * Write the data file used by the BuildTransformer
    */
-  public static async _setupJestAsync(
-    scopedLogger: ScopedLogger,
+  private async _setupJestAsync(
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    debugMode: boolean,
-    buildStageProperties: IBuildStageProperties
+    options?: IJestPluginOptions
   ): Promise<void> {
-    // Write the data file used by jest-build-transform
+    const typeScriptConfigurationJson: ITypeScriptConfigurationJson | undefined =
+      await loadTypeScriptConfigurationFileAsync(heftConfiguration, taskSession.logger.terminal);
+    const partialTsconfigFile: IPartialTsconfig | undefined = await loadPartialTsconfigFileAsync(
+      heftConfiguration,
+      taskSession.logger.terminal,
+      typeScriptConfigurationJson
+    );
+
+    // Validate, and write the Jest data file used by the BuildTransformer
     await HeftJestDataFile.saveForProjectAsync(heftConfiguration.buildFolder, {
-      emitFolderNameForTests: buildStageProperties.emitFolderNameForTests || 'lib',
-      extensionForTests: buildStageProperties.emitExtensionForTests || '.js',
-      skipTimestampCheck: !buildStageProperties.watchMode,
-      // If the property isn't defined, assume it's a not a TypeScript project since this
-      // value should be set by the Heft TypeScriptPlugin during the compile hook
-      isTypeScriptProject: !!buildStageProperties.isTypeScriptProject
+      // Use as defaults for now.
+      folderNameForTests: options?.folderNameForTests || 'lib',
+      extensionForTests:
+        options?.extensionForTests || typeScriptConfigurationJson?.emitCjsExtensionForCommonJS
+          ? '.cjs'
+          : '.js',
+      isTypeScriptProject: !!partialTsconfigFile,
+      // TODO: Handle for watch mode
+      skipTimestampCheck: true
     });
-    scopedLogger.terminal.writeVerboseLine('Wrote heft-jest-data.json file');
+    taskSession.logger.terminal.writeVerboseLine('Wrote heft-jest-data.json file');
   }
 
   /**
    * Runs Jest using the provided options.
    */
-  public static async _runJestAsync(
-    scopedLogger: ScopedLogger,
+  private async _runJestAsync(
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    debugMode: boolean,
-    testStageProperties: ITestStageProperties,
     options?: IJestPluginOptions
   ): Promise<void> {
-    const terminal: ITerminal = scopedLogger.terminal;
```

Comment:
Extract `taskSession.logger` into a variable so you don't have to keep saying `taskSession.logger`